infoschema/slow_log.go

```diff
@@ -88,6 +90,11 @@ func ParseSlowLog(tz *time.Location, scanner *bufio.Scanner) ([][]types.Datum, e
 	startFlag := false
 	var st *slowQueryTuple
 	var err error
+	// Adjust buffer size.
+	if atomic.LoadUint64(&config.QueryLogMaxLenRecord) > (bufio.MaxScanTokenSize - 100) {
+		maxBuf := int(atomic.LoadUint64(&config.QueryLogMaxLenRecord) + 100)
+		scanner.Buffer([]byte{}, maxBuf)
+	}
```

Comment:
Can we use `bufio.Reader.ReadLine` to completely avoid this error?

Reply:
We can use ReadLine to implement a new scanner

Reply:
Great idea. Other question, should we add a max-single-line-length-limit? If user executes a long query. 

At first, I think if we don't limit the max-single-line-length, then parse slow log may consume a lot of memory and  then may make TiDB-server killed by the system.

But after careful consideration, I think no need to add max-single-line-length, the first reason is we have log rotate, the other reason is executing the long query may make tidb-server killed in executing. 😂  What do you think? @eurekaka @winkyao

Reply:
https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_max_allowed_packet should limit the input query size?

Reply:
done. PTAL @eurekaka @winkyao